src/askui/telemetry/telemetry.py

```diff
@@ -0,0 +1,229 @@
+import inspect
+import os
+import platform
+import time
+from functools import wraps
+from typing import Any, Callable
+import uuid
+
+import machineid
+from pydantic import BaseModel, Field
+
+from askui.logger import logger
+from askui.telemetry.context import (
+    CallStack,
+    DeviceContext,
+    TelemetryContext,
+    AppContext,
+    OSContext,
+    PlatformContext,
+)
+from askui.telemetry.pkg_version import get_pkg_version
+from askui.telemetry.processors import SegmentSettings, TelemetryProcessor
+from askui.telemetry.user_identification import (
+    UserIdentification,
+    UserIdentificationSettings,
+)
+from askui.telemetry.utils import map_guid_to_uuid4
+
+
+class TelemetrySettings(BaseModel):
+    """Settings for telemetry configuration"""
+
+    user_identification: UserIdentificationSettings | None = (
+        UserIdentificationSettings()
+    )
+    segment: SegmentSettings | None = SegmentSettings()
+    app_name: str = "askui-vision-agent"
+    app_version: str = get_pkg_version()
+    group_id: str | None = Field(
+        default=os.environ.get("ASKUI_WORKSPACE_ID"),
+        description=(
+            'The group ID of the user. Defaults to the "ASKUI_WORKSPACE_ID" environment variable if set, '
+            "otherwise `None`."
+        ),
+    )
+    device_id: str = Field(
+        default_factory=lambda: map_guid_to_uuid4(machineid.hashed_id("askui")),
```

Comment:
Are you using the same as @mlikasam-askui? In some cases the OS function is deactivated. So the machineid can throw an exception. How do you handle this?

Reply:
Already discussed with @mlikasam-askui. Currently, not using the same. Chose a more privacy friendly approach here and moved making it consistent across components to https://askyourui.atlassian.net/browse/CL-1317 

Good catch. Came up with fallback. I have another talk with @mlikasam-askui about this as I touch `~/.askui`